forms-web-app/src/controllers/appellant-submission/task-list.js

```diff
@@ -1,119 +1,80 @@
+const { getTaskStatus } = require('../../services/task.service');
+const { SECTIONS } = require('../../services/task.service');
 const { VIEW } = require('../../lib/views');
 
+function countTasks(sections) {
+  let nbTasks = 0;
+  let nbCompleted = 0;
+
+  sections.forEach((section) => {
+    nbTasks += section.items.length;
+    nbCompleted += section.items.filter((subSection) => subSection.status === 'COMPLETED').length;
+  });
+
+  return { nbTasks, nbCompleted };
+}
+
+const HEADERS = {
+  aboutYouSection: 'About you',
+  yourDetails: 'Your details',
+  requiredDocumentsSection: 'About the original planning application',
+  applicationNumber: 'Planning application number',
+  originalApplication: 'Upload the original planning application form',
+  decisionLetter: 'Upload the decision letter',
+  dateApplied: 'Date you applied for planning permission',
+  yourAppealSection: 'Your appeal',
+  appealStatement: 'Your appeal statement',
+  otherDocuments: 'Any other documents to support your appeal',
+  otherAppeals: 'Other relevant appeals',
+  appealSiteSection: 'About the appeal site',
+  siteAddress: 'Address of the appeal site',
+  siteAccess: 'Access to the appeal site',
+  siteOwnership: 'Ownership of the appeal site',
+  healthAndSafety: 'Any health and safety issues',
+  submitYourAppealSection: 'Submit your appeal',
+  checkYourAnswers: 'Check your answers',
+};
+
+function buildTaskLists(appeal) {
+  const taskList = [];
+  const sections = SECTIONS;
+  Object.keys(sections).forEach((sectionName) => {
+    const section = sections[sectionName];
+
+    const task = {
+      heading: {
+        text: HEADERS[sectionName],
+      },
+      items: [],
+    };
+
+    Object.keys(section).forEach((subSectionName) => {
+      const subSection = section[subSectionName];
+
+      const status = getTaskStatus(appeal, sectionName, subSectionName);
+      task.items.push({
+        text: HEADERS[subSectionName],
+        href: subSection.href,
+        // attributes: { status }, e2e-tests
+        status,
+      });
+    });
+    taskList.push(task);
+  });
+  return taskList;
+}
```

Comment:
feels like lines 5 to here could be in a separate module - `lib/count-tasks` or something? 

I appreciate it's all related to this controller, just fries my brain when there's lots to look at in any given file

Reply:
I understand, and I completely agree that there is refactoring to do. I also feel that task.service is too big even if I don't see a more logical way and future proof way to do this at the moment.  

I'm pretty sure that at some point we will need to do partial appeal update and we'll be able to make the hole thing more modular and nice. That's why I didn't want to split too much